packages/data/src/registry.js

```diff
@@ -251,7 +291,31 @@ export function createRegistry( storeConfigs = {} ) {
 			throw new TypeError( 'Must specify store reducer' );
 		}
 
-		const store = registerReducer( reducerKey, options.reducer, options.persist );
+		// REMOVEME: Deprecation: v3.7
+		if ( options.reducer.__keyToPersist ) {
+			options.persist = [ options.reducer.__keyToPersist ];
+		}
+
+		if ( options.middlewares ) {
```

Comment:
I'm personally not certain we'd like to expose `middlewares`/`enhancers` because these are very highly tied to Redux and I know there ware discussions about changing enhancers API in future versions of Redux.

Reply:
>I know there ware discussions about changing enhancers API in future versions of Redux.

Do you have a link for more information?

Enhancers aren't too much more than what we already enable by returning the store from `registerStore`. We're using this ourselves to apply middleware manually, so it's obviously a requirement already.

The main goal here was avoiding the need to expose `namespaces`, which is an even deeper implementation detail that I'd worried to expose. Authoring as a middleware was a means to reimplementing persistence to avoid depending on it, as an alternative to checking whether state has changed after a dispatch. That said, it could also be done with `subscribe` now that I think of it.

I don't feel too strongly either way, but I also don't think middlewares / enhancers are too "internal" or "Redux"-y to warrant hiding; that said, I'm curious about the discussion you refer to. 😄

Reply:
Skimming the discussion there, I can see how most everything could be implemented on a base of `getState` and `dispatch`. I even think in the pull request there that it could be taken further:

- `onChange` is an unnecessary part of `storeBase` which can be applied as composed in overriding `dispatch`.
- `initialState` could be avoided, assigned in the reducer as the default value for the state argument.
   - This loses parallels to [Array#reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce), but I assume an intention with `initialValue` there is in allowing intentional accumulation of `undefined`, which is forbidden in Redux.

There's an open question of convenience and that these patterns are already well-established in the Redux ecosystem.

For the purposes of this pull request, I can see whether it's easy enough to avoid the enhancers by just overriding dispatch on the registered store instance.